feat(HMS-2994): delete domain confirmation

avisiedo opened the pull request:
This change introduce the delete confirmation modal
dialog.
    
This dialog will show up when we click on the kebab
menu of the list domains view to delete the item, or
when we click on the kebab menu from the detail view
of the domain to delete the domain that is being
displayed.
    
In both cases, the deletion confirmation modal is
showed up, and when we press Delete button the
item is eventually deleted from the database.
    
Making click on the modal cross icon or cancel link
will dismiss the deletion confirmation modal with
no effect on the data stored.

pvoborni commented on src/Components/ConfirmDeleteDomain/ConfirmDeleteDomain.test.tsx:
```diff
@@ -0,0 +1,31 @@
+      <>
+        Delete identity domain registration?
+        <br />
+        Now new host enrollment from HCC will be allowed on <b>{domain.domain_name}</b> domain after registration deletion.
```
Should the displayed domain name be field: `domain.domain_name` or `domain.title` ?  I.e. with `domain_name` the displayed name is different with the value in column "name" in the domain list.

For this reason I've used `title` in https://github.com/podengo-project/idmsvc-frontend/pull/50/files#diff-487d23f05b405c8a325218f25d07497a626079b047abe5a3f4b4b832e2d172c5R49

We should agree on the same to use in both dialogs - to be consistent.

avisiedo replied:
Thanks! I have updated to use `domain.title` to be consistent :+1:

pvoborni commented on src/Routes/DetailPage/DetailPage.tsx:
```diff
@@ -82,26 +84,38 @@ const DetailPage = () => {
     setIsKebabOpen(!isKebabOpen);
   };
 
+  const OnShowConfirmDelete = (_domain: Domain) => {
```
The `_domain` param can be removed as it is not used. It will also fix the linter warning.

avisiedo replied:
thanks! I have removed the parameter.

pvoborni reviewed (commented):
Works well. Mostly LGTM, but see the Q and comment.

avisiedo commented:
I have updated unit tests to be more accurate; the check for the text content of the modal was not working as I expected, so I have removed to avoid confusion; if some immediate way to add the check, happy to know about it.

pvoborni commented on src/Routes/DetailPage/DetailPage.tsx:
```diff
@@ -82,26 +84,38 @@ const DetailPage = () => {
-            .then((res) => {
-              if (res.status === 204) {
-                console.info('Domain ' + value + ' was deleted');
-                appContext?.deleteDomain(domain_id);
```
Will rather ask. `appContext?.deleteDomain(domain_id);` is not part of the new `onDelete.` Is it intentional?

Missing it is not a big deal. There is a period of time after redirection when the deleted domain is still visible, but it gets corrected automatically when the domains list page refreshes its content. The refresh took some time in my env when I had there 12 domains.

avisiedo replied:
thanks! good catch! I didn't realize! :+1: 

I am updating the code.

pvoborni replied:
Thanks - LGTM again.

pvoborni reviewed (approved):
It works. Code LGTM. Approving, but see the comment.
